datamations/datamation_groupby.py

```diff
@@ -466,73 +505,100 @@ def prep_specs_summarize(self, width=300, height=300):
+                id = id + 1
 
         if self.operations[-1] == "mean":
+            meta = {
+                "axes": len(self._by) > 1,
+                "description": "Plot " + self.operations[-1] + " " + y_axis + " of each group, with errorbar"
+            }
             spec_encoding = { 'x': x_encoding, 'y': y_encoding, 'tooltip': tooltip }
             if len(self._by) > 1:
                 spec_encoding = { 'x': x_encoding, 'y': y_encoding, "color": color, 'tooltip': tooltip }
             spec = utils.generate_vega_specs(data, meta, spec_encoding, facet_encoding, facet_dims, True)
             specs_list.append(spec)
 
         # Show the summarized values along with error bars, zoomed in
-        if len(self._by) > 2:
-            min_array = []
-            max_array = []
-            for group in groups:
-                for subgroup in subgroups:
-                    for l3group in l3groups:
-                        if l3group == "NA":
-                            l3group = float("nan")
-                        if subgroup in self._output[y_axis][group] and l3group in self._output[y_axis][group][subgroup]:
-                            if self.operations[-1] == "mean":
-                                min_array.append(self._output[y_axis][group][subgroup][l3group] - self._error[y_axis][group][subgroup][l3group])
-                            else:
-                                min_array.append(self._output[y_axis][group][subgroup][l3group])
-                        if subgroup in self._output[y_axis][group] and l3group in self._output[y_axis][group][subgroup]:
-                            if self.operations[-1] == "mean":
-                                max_array.append(self._output[y_axis][group][subgroup][l3group] + self._error[y_axis][group][subgroup][l3group])
-                            else:
-                                max_array.append(self._output[y_axis][group][subgroup][l3group])
-            domain = [round(min(min_array), 13), round(max(max_array), 13)]
-        elif len(self._by) > 1:
-            domain = [
-                round(min(self._output[y_axis][groups[0]][subgroups[0]] - self._error[y_axis][groups[0]][subgroups[0]], self._output[y_axis][groups[1]][subgroups[1]] - self._error[y_axis][groups[0]][subgroups[0]]),13),
-                round(max(self._output[y_axis][groups[1]][subgroups[0]] + self._error[y_axis][groups[1]][subgroups[0]], self._output[y_axis][groups[1]][subgroups[1]] + self._error[y_axis][groups[1]][subgroups[1]]),13),
-              ]
-        else:
-            domain = [
-                round(min(self._output[y_axis][groups[0]]-self._error[y_axis][groups[0]], self._output[y_axis][groups[1]]-self._error[y_axis][groups[1]]),13),
-                round(max(self._output[y_axis][groups[0]]+self._error[y_axis][groups[0]], self._output[y_axis][groups[1]]+self._error[y_axis][groups[1]]),13)
-            ]
+        if any(element in ['mean', 'median'] for element in self.operations):
```

Comment:
This condition should go after the length check?

Reply:
Are you referring to the length check that comes directly after that condition? If you are, wouldn't I need to include this condition inside all the elif and else statements connected to the first length check?